src/pages/index.js

```diff
@@ -22,128 +23,261 @@ import {
+  gallerySelector
+);
+
+api
+  .getInitialCards()
+  .then((res) => {
+    if (res.length === 0) {
+      // If no cards are in the database
+      // add initialCards to server
+      initialCards.forEach((card) => {
+        api.addNewCard(card);
+      });
+    }
+
+    res.forEach((item) => {
+      // create new card
+      const newCard = createCard(item, openImageModal);
+      // Add new card to begining of card gallery
+      cardsSection.addItem(newCard);
+    });
+
+    cardsSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+//// Functions ////
 function fillProfileForm() {
   const userInfo = user.getUserInfo();
   profilePopup.setInputValues(userInfo);
+}
 
-  // profileNameInput.value = userInfo.name;
-  // profileDescriptionInput.value = userInfo.description;
+function openEditAvatarModal() {
+  editAvatarPopup.open();
 }
 
-// Button Handlers
 function openEditProfileModal() {
   fillProfileForm();
   formValidators[profileForm.getAttribute("id")].resetValidation();
 
   profilePopup.open();
 }
 
+function openImageModal(data) {
+  imgagePopup.open(data);
+}
+
 function openAddCardModal() {
   addCardPopup.open();
 }
 
-// Button Event Listeners
+function createCard(data) {
+  const cardElement = new Card(
+    data,
+    cardTemplate,
+    openImageModal,
+    openDeleteModal,
+    handleLikeButton
+  );
+
+  return cardElement.getView();
+}
+
+function openDeleteModal(currentCardId, currentCardElement) {
+  deleteCardPopup.setButtonText("Yes");
+  deleteCardPopup.open();
+  deleteCardPopup.currentCardId = currentCardId;
+  deleteCardPopup.currentCardElement = currentCardElement;
+}
+
+function handleCardDelete() {
+  deleteCardPopup.currentCardElement.remove();
+
+  return api.deleteCard(deleteCardPopup.currentCardId).catch((err) => {
+    console.error("Delete Card Error:", err);
+  });
+}
+
+function handleLikeButton(cardId, isLiked) {
+  console.log(cardId);
+  console.log(isLiked);
+
+  if (!isLiked) {
+    return api
+      .addLike(cardId)
+      .then(() => true)
+      .catch((err) => {
+        console.error("Like Button Error:", err);
+        return false;
+      });
+  } else {
+    return api
+      .removeLike(cardId)
+      .then(() => false)
+      .catch((err) => {
+        console.error("Like Button Error:", err);
+        return true;
+      });
+  }
+}
+
+//// Button Event Listeners ////
+// avatar image click
+profileAvatarWrapper.addEventListener("click", openEditAvatarModal);
+
 // profile edit button click
 profileEditButton.addEventListener("click", openEditProfileModal);
 
 // add card button click
 addCardButton.addEventListener("click", openAddCardModal);
 
-//// Forms ////
-const profileForm = document.forms["edit-profile-form"];
-const addCardForm = document.forms["add-card-form"];
+//// Form Submit Handlers ////
+// Avatar form submission handler
+function handleAvatarFormSubmit(data) {
+  profileAvatar.src = data.link;
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error, it shouldn’t change